### PR TITLE
feat: add terradoc

### PR DESCRIPTION
**Pull request by kassianh**
Recreated README with terradoc, checked documentation for errors or inconsistencies.

### CODE REVIEW

File: README.tfdoc.hcl [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+    title   = "Module Argument Reference"
+    content = <<-END
+      See [variables.tf] and [examples/] for details and use-cases.
+    END
+
+    section {
+      title = "Top-level Arguments"
+
+      section {
+        title = "Module Configuration"
```
Comment (mariux) [inline README.tfdoc.hcl:86]: module config should be last section in "Module Argument Reference"....  also just noticed as we use nested attributes now top-level arguments does not make any sense anymore... will create a PR to adjust the template.... my-bad that this was missed so far..

File: README.md [hunk 2]
```diff
@@ -158,28 +167,29 @@ See [variables.tf] and [examples/] for details and use-cases.
...
+  - [**`next_rotation_time`**](#attr-rotation-next_rotation_time): *(Optional `string`)*<a name="attr-rotation-next_rotation_time"></a>
 
     Timestamp in UTC at which the Secret is scheduled to rotate. A timestamp in RFC3339 UTC `Zulu` format, with nanosecond resolution and up to nine fractional digits. Examples: `2014-10-02T15:01:23Z` and `2014-10-02T15:01:23.045123456Z`.
 
-  - **`rotation_period`**: _(Optional `string`)_
+  - [**`rotation_period`**](#attr-rotation-rotation_period): *(Optional `string`)*<a name="attr-rotation-rotation_period"></a>
 
-    The Duration between rotation notifications. Must be in seconds and at least 3600s (1h) and at most 3153600000s (100 years). If rotationPeriod is set, `next_rotation_time` must be set. `next_rotation_time` will be advanced by this period when the service automatically sends rotation notifications.
+    The Duration between rotation notifications. Must be in seconds and at least `3600s` (1h) and at most `3153600000s` (100 years). If rotationPeriod is set, `next_rotation_time` must be set. `next_rotation_time` will be advanced by this period when the service automatically sends rotation notifications.
```
Comment (mariux) [inline README.md:178]: very nice ;)

**Review by mariux — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -40,116 +44,98 @@ This module implements the following terraform resources:
 
 and supports additional features of the following modules:
 
-- [mineiros-io/terraform-google-secret-manager-iam](https://github.com/mineiros-io/terraform-google-secret-manager-iam)
```
Comment (mariux) [inline README.md:43]: this should stay as we are using the module

**Review by mariux — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (kassianh) [reply README.md]: ah yes, my bad.

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**